### ISSUE TITLE
Skip step

### ISSUE THREAD

**Issue by ycs77** [ycs77/laravel-wizard#11]

**Comment by veganista** [comment]
I have a similar library to yours and implemented step skipping by introducing a skip() method on the step that would return true or false depending on whether the step should be skipped.

I've added a skip() method to your abstract Step class and updated your prev() and get() methods in StepRepository to:

```php
    public function prev()
    {
        $index = $this
            ->steps
            ->reject
            ->skip()
            ->search(function($step, $index) {
                return $index < $this->currentIndex;
            });

        if ($index === false) {
            return $this->get($this->currentIndex - 1);
        }

        return $this->get($index);
        // return $this->get($this->currentIndex - 1);
    }

    public function next()
    {
        $index = $this
            ->steps
            ->reject
            ->skip()
            ->search(function($step, $index) {
                return $index > $this->currentIndex;
            });

        if ($index === false) {
            return $this->get($this->currentIndex + 1);
        }

        return $this->get($index);
        // return $this->currentIndex + 1;
    }
```

This seems to work with skipping steps. I'll try and put some more time into this if you would like to test and tidy it up.

Just thought I'd share what I'd done previously.

**Comment by ycs77** [comment]
@veganista Thank you for your suggestions, I will use as a reference.

**Comment by ycs77** [comment]
This feature will appear in the next version. (But it takes some time...)

**Comment by ycs77** [comment]
Released [v2.3.3](https://github.com/ycs77/laravel-wizard/releases/tag/v2.3.3) added skip feature.
